### PR TITLE
HTTP client moved under the respective thread

### PR DESCRIPTION
**Pull request by seran**

### CODE REVIEW

**Comment by seran** [comment]
```
java.lang.IllegalStateException: Client instance has been closed.
	at org.glassfish.jersey.internal.guava.Preconditions.checkState(Preconditions.java:169)
	at org.glassfish.jersey.client.JerseyClient.checkNotClosed(JerseyClient.java:257)
	at org.glassfish.jersey.client.JerseyClient.target(JerseyClient.java:272)
	at org.glassfish.jersey.client.JerseyClient.target(JerseyClient.java:56)
	at org.evomaster.core.problem.externalservice.httpws.service.HarvestActualHttpWsResponseHandler.buildInvocation(HarvestActualHttpWsResponseHandler.kt:271)
	at org.evomaster.core.problem.externalservice.httpws.service.HarvestActualHttpWsResponseHandler.createInvocationToRealExternalService(HarvestActualHttpWsResponseHandler.kt:304)
	at org.evomaster.core.problem.externalservice.httpws.service.HarvestActualHttpWsResponseHandler.sendRequestToRealExternalService(HarvestActualHttpWsResponseHandler.kt:169)
	at org.evomaster.core.problem.externalservice.httpws.service.HarvestActualHttpWsResponseHandler.addRequest$lambda$3(HarvestActualHttpWsResponseHandler.kt:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by seran** [comment]
```
Exception in thread "pool-3-thread-2" java.lang.OutOfMemoryError: Java heap space
	at java.util.Arrays.copyOf(Arrays.java:3332)
	at java.lang.AbstractStringBuilder.ensureCapacityInternal(AbstractStringBuilder.java:124)
	at java.lang.AbstractStringBuilder.append(AbstractStringBuilder.java:596)
	at java.lang.StringBuilder.append(StringBuilder.java:190)
	at org.glassfish.jersey.message.internal.ReaderWriter.readFromAsString(ReaderWriter.java:153)
	at org.glassfish.jersey.message.internal.ReaderWriter.readFromAsString(ReaderWriter.java:137)
	at org.glassfish.jersey.message.internal.AbstractMessageReaderWriterProvider.readFromAsString(AbstractMessageReaderWriterProvider.java:94)
	at org.glassfish.jersey.message.internal.StringMessageProvider.readFrom(StringMessageProvider.java:54)
	at org.glassfish.jersey.message.internal.StringMessageProvider.readFrom(StringMessageProvider.java:36)
	at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$TerminalReaderInterceptor.invokeReadFrom(ReaderInterceptorExecutor.java:233)
	at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$TerminalReaderInterceptor.aroundReadFrom(ReaderInterceptorExecutor.java:212)
	at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor.proceed(ReaderInterceptorExecutor.java:132)
	at org.glassfish.jersey.message.internal.MessageBodyFactory.readFrom(MessageBodyFactory.java:1072)
	at org.glassfish.jersey.message.internal.InboundMessageContext.readEntity(InboundMessageContext.java:885)
	at org.glassfish.jersey.message.internal.InboundMessageContext.readEntity(InboundMessageContext.java:819)
	at org.glassfish.jersey.client.ClientResponse.readEntity(ClientResponse.java:298)
	at org.glassfish.jersey.client.InboundJaxrsResponse$1.call(InboundJaxrsResponse.java:93)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:205)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:365)
	at org.glassfish.jersey.client.InboundJaxrsResponse.runInScopeIfPossible(InboundJaxrsResponse.java:244)
	at org.glassfish.jersey.client.InboundJaxrsResponse.readEntity(InboundJaxrsResponse.java:90)
	at org.evomaster.core.problem.externalservice.httpws.service.HarvestActualHttpWsResponseHandler.handleActualResponse(HarvestActualHttpWsResponseHandler.kt:353)
	at org.evomaster.core.problem.externalservice.httpws.service.HarvestActualHttpWsResponseHandler.sendRequestToRealExternalService(HarvestActualHttpWsResponseHandler.kt:168)
	at org.evomaster.core.problem.externalservice.httpws.service.HarvestActualHttpWsResponseHandler.addRequest$lambda$3(HarvestActualHttpWsResponseHandler.kt:266)
	at org.evomaster.core.problem.externalservice.httpws.service.HarvestActualHttpWsResponseHandler$$Lambda$257/651535127.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by seran** [comment]
Moving the HTTP client under the respective thread eradicated these exceptions.

File: core/src/main/kotlin/org/evomaster/core/problem/externalservice/httpws/service/HarvestActualHttpWsResponseHandler.kt [hunk 1]
```diff
@@ -267,7 +249,14 @@ class HarvestActualHttpWsResponseHandler {
         workerPool.execute(task)
     }
 
-    private fun buildInvocation(httpRequest: HttpExternalServiceRequest): Invocation {
+    private fun fetchActualResponse(httpRequest: HttpExternalServiceRequest): ActualResponseInfo? {
+        val clientConfiguration = getClientConfiguration()
+
+        val httpWsClient = ClientBuilder.newBuilder()
```
Comment (man-zhang) [inline core/src/main/kotlin/org/evomaster/core/problem/externalservice/httpws/service/HarvestActualHttpWsResponseHandler.kt:255]: cannot create a client for each request. 
I will try to handle this task.

**Pull request closed without merging**